In [19]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

from sklearn.datasets import load_digits
digits = load_digits()
# digits

x = digits.data
y = digits.target
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size = 0.2)

from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier

model_params = {
    "logistic regression":{"model":LogisticRegression(),
          "params":{"C":[1,5,10]}},
    "decision tree":{"model":DecisionTreeClassifier(),
           "params":{"criterion":["gini","entropy"]}},
    "svm":{"model":SVC(gamma = "auto"),
           "params":{"C":[1,10,20],"kernel":["rbf","linear"]}},
    "random forest":{"model":RandomForestClassifier(),
           "params":{"n_estimators":[1,5,10]}}
}

from sklearn.model_selection import GridSearchCV
clf = GridSearchCV(SVC(gamma = "auto"),
                   {"C":[1,10,20],"kernel":["rbf","linear"]},
                   cv = 5,return_train_score = False)
clf.fit(x_train,y_train)
clf.cv_results_
# clf.best_score_
# clf.best_params_

scores = []
for key,value in model_params.items():
    clf = GridSearchCV(value["model"],value["params"],
                       cv = 5,return_train_score = False)
    clf.fit(x_train,y_train)
    scores.append({"model":key,
                   "best_score":clf.best_score_,
                   "best_params":clf.best_params_})
df = pd.DataFrame(scores)
df

model  best_score                   best_params
0  logistic regression    0.966609                      {'C': 1}
1        decision tree    0.868474      {'criterion': 'entropy'}
2                  svm    0.978431  {'C': 1, 'kernel': 'linear'}
3        random forest    0.938753          {'n_estimators': 10}

- For me, the winner is Support Vector Machine(SVM) and the parameters are C=1, kernel=linear with 97.84% score.